### PR TITLE
DTSPO-4675 - system app tolerations

### PR DESCRIPTION
**Pull request by msl8r**
### Change description ###
- set tolerations and node selector for system apps 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: k8s/namespaces/admin/aad-pod-identity/patches/sbox/aad-pod-id.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        key: CriticalAddonsOnly
+        operator: Equal
+        value: "true"
+      affinity:
+        nodeAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 1
+            preference:
+              matchExpressions:
+              - key: agentpool
```
Comment (timja) [inline k8s/namespaces/admin/aad-pod-identity/patches/sbox/aad-pod-id.yaml:42]: isn't it a full domain key not just the text agentpool ?

i.e. we should probably use the label and not the agentpool name

Comment (timja) [reply k8s/namespaces/admin/aad-pod-identity/patches/sbox/aad-pod-id.yaml:42]:
```suggestion
              - key: kubernetes.azure.com/mode
```

Comment (msl8r) [reply k8s/namespaces/admin/aad-pod-identity/patches/sbox/aad-pod-id.yaml:42]: yeah I started doing that then got distracted 🤦🏼, will do the rest now

Comment (msl8r) [reply k8s/namespaces/admin/aad-pod-identity/patches/sbox/aad-pod-id.yaml]: good spot, updated 👍🏼

File: k8s/release/admin/secrets-csi-driver/patches/nonprod/common/secrets-csi-driver.yaml [hunk 2]
```diff
@@ -7,4 +7,18 @@ metadata:
 spec:
   releaseName: csi-secrets-store-provider-azure
   chart:
-    version: 0.2.0
\ No newline at end of file
+    version: 0.2.0
+  values:
+    secrets-store-csi-driver:
```
Comment (timja) [inline k8s/release/admin/secrets-csi-driver/patches/nonprod/common/secrets-csi-driver.yaml:12]: no node affinity?

Comment (msl8r) [reply k8s/release/admin/secrets-csi-driver/patches/nonprod/common/secrets-csi-driver.yaml:12]: nah they're both daemonsets

**Review by timja — APPROVED** [review]

**Pull request merged**